src/multihash.rs

```diff
@@ -173,6 +173,45 @@ impl<const S: usize> Multihash<S> {
             .expect("writing to a vec should never fail");
         bytes
     }
+
+    /// Truncates the multihash to the given size. It's up to the caller to ensure that the new size
+    /// is safe to use.
+    ///
+    /// If the new size is larger than the current size, this method does nothing.
+    ///
+    /// ```
+    /// use multihash::{Code, MultihashDigest};
+    ///
+    /// let hash = Code::Sha3_256.digest(b"Hello world!").truncate(20);
+    /// ```
+    pub fn truncate(&self, size: u8) -> Self {
+        let mut mh = *self;
+        mh.size = mh.size.min(size);
+        mh
+    }
+
+    /// Resizes the backing multihash buffer. This function fails if the hash digest is larger than
+    /// the target size.
+    ///
+    /// ```
+    /// use multihash::{Code, MultihashDigest, MultihashGeneric};
+    ///
+    /// let hash = Code::Sha3_256.digest(b"Hello world!");
+    /// let large_hash: MultihashGeneric<32> = hash.resize().unwrap();
+    /// ```
+    pub fn resize<const R: usize>(&self) -> Result<Multihash<R>, Error> {
```

Comment:
I wonder if grow would be a more fitting name, given that it can only be larger than the original